src/content_mapblock.cpp

```diff
@@ -172,6 +172,10 @@ static inline void getNeighborConnectingFace(v3s16 p, INodeDefManager *nodedef,
 		*neighbors |= v;
 }
 
+static int NeighbourToIndex(const v3s16 &pos) {
+	return (9 * pos.X + 3 * pos.Y + pos.Z) + 13;
```

Comment:
13 is used to make the return value >= 0

pos.X ∈ {-1; 0; 1}
pos.Y ∈ {-1; 0; 1}
pos.Z ∈ {-1; 0; 1}
NeighbourToIndex(pos) ∈ {0; 1; …; 26}